### PR TITLE
Fix affected acct

### PR DESCRIPTION
**Pull request by kimmylin**
- make sure coinbases are included in affected accounts
- make sure fee payers (calculated from block txs) are included in affected accounts
- make sure tx fees are considered in total balance regardless an event is created

### CODE REVIEW

**Comment by kimmylin** [comment]
screen shot 2018-07-13 at 9 55 16 am
the old code misses block number 5817941 that has a fee payment but no event. the new code now finds it

**Comment by kimmylin** [comment]
original bug where block 5817941 is not recorded for this account.
screen shot 2018-07-11 at 10 55 29 am

File: store/transfer_processor.go [hunk 1]
```diff
@@ -69,14 +68,33 @@ func newTransferProcessor(blockNumber int64,
...
 	// Update total balance for new subscriptions, map[group][token]balance
 	totalBalances := make(map[int64]map[gethCommon.Address]*big.Int)
 	totalFees := make(map[int64]*big.Int)
 
 	// Collect modified addresses
 	seenAddrs := make(map[gethCommon.Address]struct{})
 	var addrs [][]byte
+	// From block / uncles coinbases
+	for _, cb := range coinbases {
+		s.logger.Info("coinbase", "cb", cb.Hex())
```
Comment (markya0616) [inline store/transfer_processor.go]: Remove this log

File: store/transfer_processor.go [hunk 2]
```diff
@@ -163,22 +170,46 @@ func (s *transferProcessor) process(ctx context.Context, events []*model.Transfe
...
 		}
 		if hasTo {
 			to := gethCommon.BytesToAddress(e.To)
 			contractsAddrs[contractAddr][to] = struct{}{}
 		}
 	}
 
+	// Make sure coinbases are also included in balance queries to geth
+	for _, addr := range coinbases {
+		if allSubs[addr] != nil {
```
Comment (markya0616) [inline store/transfer_processor.go]: How about 
```
if allSubs[addr] == nil {
   continue
}
...
```

File: store/transfer_processor.go [hunk 3]
```diff
@@ -163,22 +170,46 @@ func (s *transferProcessor) process(ctx context.Context, events []*model.Transfe
...
+			contractsAddrs[model.ETHAddress][addr] = struct{}{}
+		}
+	}
+
+	// Collect tx fee and make sure fee payers are included in balance queries to geth
+	// Note that we don't calculate fees from events because events with transfer value 0 (but still incur fee)
+	// are not included in events.
+	feeDiff := make(map[gethCommon.Address]*big.Int)
+	for i, tx := range s.txs {
+		// Assume the tx and receipt are in the same order
```
Comment (markya0616) [inline store/transfer_processor.go]: Suggest 
```
from := gethCommon.BytesToAddress(tx.From)
if allSubs[from] == nil {
    continue
}
...
```

File: store/transfer_processor.go (hunk 1 above)
Comment (kimmylin) [reply store/transfer_processor.go]: done

File: store/transfer_processor.go (hunk 2 above)
Comment (kimmylin) [reply store/transfer_processor.go]: done

File: store/transfer_processor.go (hunk 3 above)
Comment (kimmylin) [reply store/transfer_processor.go]: done

File: migration/db/schema.rb [hunk 4]
```diff
@@ -36,6 +36,12 @@
     t.binary "extra_data", limit: 1024
     t.binary "mix_digest", limit: 32, null: false
     t.binary "nonce", limit: 8, null: false
+    t.string "miner_reward", limit: 32, null: false
```
Comment (bailantaotao) [inline migration/db/schema.rb]: We need create a migration if database table cannot be deleted on staging

Comment (bailantaotao) [reply migration/db/schema.rb]: My fault, sorry :P

**Pull request merged**